flink-clients/src/main/java/org/apache/flink/client/ClientUtils.java

```diff
@@ -105,13 +103,9 @@ public static JobExecutionResult submitJob(
 			JobGraph jobGraph) throws ProgramInvocationException {
 		checkNotNull(client);
 		checkNotNull(jobGraph);
-		try {
-			return client
-				.submitJob(jobGraph)
-				.thenApply(JobSubmissionResult::getJobID)
-				.thenApply(DetachedJobExecutionResult::new)
-				.get();
-		} catch (InterruptedException | ExecutionException e) {
+		try (JobClient jobClient = client.submitJob(jobGraph).get()){
```

Comment:
Why not closing the `ClusterClient` here?

Reply:
We don't have the ownership of `ClusterClient` here and if you take a look at its callers you will find that either the caller close `ClusterClient` immediately properly or the caller close a shared `ClusterClient` later when it is end of life.